test-data/unit/check-classvar.test

```diff
@@ -343,3 +343,18 @@ class C:
     g: ClassVar[Union[Callable[[C], int], int]] = f
 
 reveal_type(C().g)  # N: Revealed type is "Union[def () -> builtins.int, builtins.int]"
+
+[case testGenericSubclassAccessNoLeak]
+from typing import ClassVar, Generic, TypeVar
+
+T = TypeVar("T")
+class B(Generic[T]):
+    x: T
+    y: ClassVar[T]
+
+class C(B[T]): ...
```

Comment:
I think it would be good to also test attribute access in cases like `class D(B[int]): ...`. Should `D.y` have type `int` then?